src/Tempest/Generation/src/StubFileGenerator.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Generation;
+
+use Tempest\Console\Console;
+use function Tempest\get;
+use Tempest\Support\PathHelper;
+use Tempest\Support\StringHelper;
+
+/**
+ * This class can generate a file from a stub file with additional useful methods.
+ * It only works with PHP class files.
+ */
+final class StubFileGenerator
+{
+    private Console $console;
+
+    /**
+     * @param string $targetPath The path where the generated file will be saved including the filename and extension.
+     * @param class-string $stubFile The stub file to use for the generation.
+     * @param array<string, string> $replacements An array of key-value pairs to replace in the stub file.
+     *     The keys are the placeholders in the stub file (e.g. 'DummyNamespace')
+     *     The values are the replacements for the placeholders (e.g. 'App\Models')
+     * @param bool $shouldOverride Whether the generator should override the file if it already exists.
+     */
+    public function __construct(
```

Comment:
Is there a reason why this data is injected into the constructor instead of being passed to the generate method? Tempest tries to have all classes injectable, and aims to have a clear separation of data and functionality. When I see a "Stub File Generator", my assumption is that it's a class that I can inject and reuse, while its `generate` method would require me to pass in context-specific data.

Something like this:

```php
$this->stubGenerator->generate($stub, $target, $replacements, $shouldOverwrite);
```

Reply:
This is because atm the generator isn't a service but something like a DTO and the generate method is expected to be run once.
The instance is created and returned by the `GeneratorCommand` method handler.
In case, if we decide to change the overall approch this could be changed as you mention I guess 😄